persistence/javadsl/src/main/scala/com/lightbend/lagom/javadsl/persistence/PersistentEntity.scala

```diff
@@ -402,10 +402,10 @@ abstract class PersistentEntity[Command, Event, State] {
   /**
    * INTERNAL API
    */
-  private[lagom] trait PersistNone[B <: Event] extends Persist[B] {
+  private[lagom] case class PersistNone[B <: Event]() extends Persist[B] {
```

Comment:
Should it be a singleton object?

Reply:
I can't because it has / needs a type param. On the other hand, there is no reason  to have one. Instead, it should be 
```case object PersistNone extends Persist[Nothing]```

Reply:
Just checked the full implementation and the scala api version. Weird! 

The Scala API is:
```scala
trait Persist // no type param
private[lagom] case object PersistNone extends Persist
```

The Java API is: 
```scala
abstract class Persist[B <: Event]
private[lagom] case class PersistNone[B <: Event]() extends Persist[B] {
  override def toString: String = "PersistNone"
}
private val persistNone: PersistNone[Nothing] = PersistNone[Nothing]
```

Reply:
Java doesn't have bottom types, so you can't use `Nothing` in Java.

Reply:
But does this get exposed to Java? I guess not since we are using the bottom type in `private val persistNone: PersistNone[Nothing] = PersistNone[Nothing]`.

If that's being used, we could also have: 
```scala
case object PersistNone extends PersistNone[Nothing]
```
Anyway, this is all marked as private and I don't think they are being using by any internal Java class. 

And the Scala version is not using the bottom type because the trait has no type param.